samples/model_repository/vllm_opt/config.pbtxt

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+name: "vllm_opt"
+backend: "vllm"
+
+# Disabling batching in Triton, let vLLM handle the batching on its own.
+max_batch_size: 0
+
+# We need to use decoupled transaction policy for saturating
+# vLLM engine for max throughtput.
+# TODO [DLIS:5233]: Allow asynchronous execution to lift this
+# restriction for cases there is exactly a single response to
+# a single request.
+model_transaction_policy {
+  decoupled: True
+}
+
+input [
+  {
+    name: "PROMPT"
+    data_type: TYPE_STRING
+    dims: [ 1 ]
+  },
+  {
+    name: "STREAM"
+    data_type: TYPE_BOOL
+    dims: [ 1 ]
+  },
+  {
+    name: "SAMPLING_PARAMETERS"
+    data_type: TYPE_STRING
+    dims: [ 1 ]
+    optional: true
+  }
+]
+
+output [
+  {
+    name: "TEXT"
+    data_type: TYPE_STRING
+    dims: [ -1 ]
+  }
+]
```

Comment:
I think we should consider moving I/O to python model autocomplete, because these should be fixed for the python-based backend. It's unintuitive to have users copy this for every config they write. CC @dyastremsky @tanmayv25 @pskiran1 @oandreeva-nv @nnshah1

Reply:
Are they always fixed? I thought Python-based backends can be any backend that's Python-based rather than C++-based, so they can have any inputs and outputs. They would be super flexible backends, equivalent to how Python models are super flexible (user-defined models).

These are great thoughts and an important discussion to have, especially since we want to maximize config autocompletion. In regards to the scope this PR, it is to create a rough draft of documentation so that users can use the vLLM backend as soon as it is out. We would probably want to create a separate ticket to look into what we can implement in regards to config autocompletion for Python-based backends. There is also a separate ticket (CC: @oandreeva-nv) for more detailed tutorials and documentation.

Reply:
I think in the move from a tutorial / sample to being a backend - we will need to autocomplete the config. In the current implementation the tensor names are not flexible (i.e. the model.py makes assumptions on the names). 
 
We can discuss if autocomplete is in scope - or needs to be addressed separately. If out of scope we will need to be careful to document that these need to be copied exactly -

Reply:
> Are they always fixed? I thought Python-based backends can be any backend that's Python-based rather than C++-based, so they can have any inputs and outputs. 

I mean for vllm specifically.

When a user writes:
```
# config.pbtxt
backend: vllm
```
using our `vllm/model.py`, the I/O tensor names are fixed and assumed. So we should remove the extra step of having to figure out the I/O names when writing the config for any python-based backends we control.

Reply:
> We can discuss if autocomplete is in scope - or needs to be addressed separately. If out of scope we will need to be careful to document that these need to be copied exactly -

I'd like to see it in 23.10 if possible. It should be a transparent addition.

Reply:
Discussed offline. Ryan provided an example of how to do this.

However, since optional inputs are not yet supported, we created a ticket to do so and add this in 23.11. We also plan to add support to autocomplete the decoupled mode configuration section. For 23.10, customers will need to copy over the config for use with their vLLM backend. I documented that no changes are needed for the model config (other than adding a custom execution path section, for customers using that approach).